test/functional/toolchains/ros/test_ament.py

```diff
@@ -0,0 +1,134 @@
+import os
+import textwrap
+
+from conan.test.utils.tools import TestClient
+
+
+def test_ament_cmake():
+    """
+    Test that the ament generator generates conan_<name> folders and place CMake files
+    in the correct path
+    """
+    client = TestClient()
+    with client.chdir("lib"):
+        conanfile1 = textwrap.dedent('''
+            import os
+            from conan import ConanFile
+            from conan.tools.files import save
+            class Recipe(ConanFile):
+                name = "lib1"
+                version = "1.0"
+                settings = "os", "compiler", "build_type", "arch"
+
+                def package(self):
+                    save(self, os.path.join(self.package_folder, "lib", "lib1"), "")
+            ''')
+        conanfile2 = textwrap.dedent('''
+            import os
+            from conan import ConanFile
+            from conan.tools.files import save
+
+            class Recipe(ConanFile):
+                name = "lib2"
+                version = "1.0"
+                settings = "os", "compiler", "build_type", "arch"
+
+                def requirements(self):
+                    self.requires('lib1/1.0')
+
+                def package(self):
+                    save(self, os.path.join(self.package_folder, "lib", "lib2"), "")
+            ''')
+        conanfile3 = textwrap.dedent('''
+            [requires]
+            lib2/1.0
+            [generators]
+            Ament
+            ''')
+        client.save({
+            "conanfile1.py": conanfile1,
+            "conanfile2.py": conanfile2,
+            "conanfile3.txt": conanfile3
+        })
+
+        client.run("create conanfile1.py")
+        client.run("create conanfile2.py")
+        client.run("install conanfile3.txt --output-folder install")
+        assert "Generating CMake files for lib2 dependency" in client.out
+        # Check direct dependencies package.xml file generation
+        conan_lib2_package_xml = os.path.join(client.current_folder, "conan_lib2", "package.xml")
+        assert os.path.exists(conan_lib2_package_xml)
+        # Check CMake files generated
+        lib2_cmake_config = os.path.join(client.current_folder, "install", "conan_lib2", "share",
+                                         "lib2", "cmake", "lib2-config.cmake")
+        assert os.path.exists(lib2_cmake_config)
+        # Check lib2's transitive dependency
+        lib1_cmake_config = os.path.join(client.current_folder, "install", "conan_lib2", "share",
+                                         "lib1", "cmake", "lib1-config.cmake")
+        assert os.path.exists(lib1_cmake_config)
```

Comment:
This doesn't look like a functional test, it can be moved to ``integration`` tests.